### PR TITLE
Fix for UTC parsing not working correctly

### PR DESCRIPTION
**Pull request by rad-pat**
Per old bug https://github.com/qooxdoo/qooxdoo/issues/7946, UTC parsing was not working correctly. Also the timezone format identifiers 'Z' & 'z' were not being processed if non-literal in the format string.

 * Use date.UTC() to correctly determine the UTC date.
 * Adds isoDateTimeTz as a format to parse timezone details in the date string, i.e. offset from UTC

### CODE REVIEW

File: source/class/qx/util/format/DateFormat.js [hunk 1]
```diff
@@ -1207,10 +1216,28 @@ qx.Class.define("qx.util.format.DateFormat",
         dateValues.hour = parseInt(value, 10) % 12;
       };
 
-      var ignoreManipulator = function(dateValues, value) {
-        return;
+      var timezoneManipulator = function(dateValues, value) {
+        var regEx = new RegExp("([+-]*)(\\d\\d)[:]*(\\d\\d)*$");
```
Comment (derrell) [inline source/class/qx/util/format/DateFormat.js:1220]: This regex doesn't look right. It allows 0 or more leading `+` or `-`, and zero or more colons. That means that this value is accepted (but `tzResults[1]` does not equal `-`):
```
"---23:::13"
```

I think you probably mean "options" (0 or 1) in both cases, so you'd use this:
```
var regEx = new RegExp("([+-]?)(\\d\\d)[:]?(\\d\\d)?$");
```

Even that, though, doesn't invalidate bad input. I think there are additional constraints necessary to detect bad input.

**Review by derrell — CHANGES_REQUESTED** [review]

File: source/class/qx/util/format/DateFormat.js (hunk 1 above)
Comment (oetiker) [reply source/class/qx/util/format/DateFormat.js:1220]: I think `var regEx = new RegExp("([+-]?)(\\d\\d)(?::?(\\d\\d))?$");` should do.

**Review by oetiker — CHANGES_REQUESTED** [review]
see regexp comment

File: source/class/qx/util/format/DateFormat.js (hunk 1 above)
Comment (rad-pat) [reply source/class/qx/util/format/DateFormat.js]: Thanks, I've updated the regex. This regex is only used for extracting the parts, not for validation. Validation is already done via regexes on lines 1784 & 1791 for rules 'Z' and 'z' respectively.

**Review by cboulanger — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by derrell — APPROVED** [review]

**Review by hkollmann — APPROVED** [review]

**Review by oetiker — APPROVED** [review]

**Pull request merged**